Modules/_asynciomodule.c

```diff
@@ -1041,8 +1055,17 @@ _asyncio_Future_remove_done_callback_impl(FutureObj *self, PyTypeObject *cls,
 
     if (len == 1) {
         PyObject *cb_tup = PyList_GET_ITEM(self->fut_callbacks, 0);
-        int cmp = PyObject_RichCompareBool(
-            PyTuple_GET_ITEM(cb_tup, 0), fn, Py_EQ);
+        // Beware: An evil PyObject_RichCompareBool could change fut_callbacks
+        // or its items (see https://github.com/python/cpython/issues/97592 or
+        // https://github.com/python/cpython/issues/125789 for details).
+        if (!PyTuple_CheckExact(cb_tup) || PyTuple_GET_SIZE(cb_tup) != 2) {
+            PyErr_SetString(PyExc_RuntimeError, "corrupted callback tuple");
+            return NULL;
+        }
+        Py_INCREF(cb_tup);
+        PyObject *cb = PyTuple_GET_ITEM(cb_tup, 0);
```

Comment:
A little nitpicky, but the incref should be on `cb` not `cb_tup`. We don't necessarily care if the tuple gets deleted in PyObject_RichCompareBool, we care if `cb` gets deleted. This technically works because I don't think there's a way to delete `cb` at this point if the tuple can't be deleted, but it's still worth noting.